packages/sdk-client/src/client.js

```diff
@@ -135,3 +142,16 @@ function compose (...funcs: Array<Function>): Function {
 
   return funcs.reduce((a, b) => (...args) => a(b(...args)))
 }
+
+function validate (request: Object, fn?: ProcessFn) {
+  if (!request)
+    throw new Error('Missing required `request` argument.')
+  if (typeof request.uri !== 'string' || !request.method)
```

Comment:
Nitpick: we could also validate that the `method` is one of the available `http` methods.

See `require('http').METHODS`